### ISSUE TITLE
Past Release sites are taking long time to load

### ISSUE THREAD

**Issue by yashcho** [jenkins-infra/helpdesk#3525]
### Service(s)

get.jenkins.io

### Summary

Below two links from page (https://www.jenkins.io/download/) are taking more than 30 seconds to load and show the table with list of old versions. Can you please look into it?

https://get.jenkins.io/war/ 

https://get.jenkins.io/war-stable/

### Reproduction steps

1. Open browser by removing cache and cookies then hit any one of link.
2. Either of them will take more than 30 second or crash with status as "502 Bad Gateway"

**Comment by dduportal** [comment]
Hi @yashcho , I can reproduce the issue, looking into it.

**Comment by dduportal** [comment]
- Triggering a node pool image upgrade to force recylcing the underlying VMs (and the associated azurefile mount) as a short ter measure

**Comment by dduportal** [comment]
- The service was upgraded, with a 2 minut outage => no more performances issues (no more timeouts)
  - Ref. https://github.com/jenkins-infra/kubernetes-management/pull/3862
- The `/war-stable` link is now working very well
  - It lists LTS releases as expected
- There is still an issue with the `/war` link: it appears that HTML page is missing some partials, work in progress
  - Direct WA download still works as expected

**Comment by dduportal** [comment]
Operations notes for the infra-team:

Preliminaries:
- A snapshot of the persistent volume was taken before operations (named `20230419-09h46-utc`)
- Checked that all involved PVs are in "retain" mode in the `mirrorbits` namespace
- Removed all the PVCs and PVs related to redis as `mirrorbits` uses an Azure-managed Redis instances since 2.5 years
  - The associated Azure disks are not deleted and should be once `prodpublick8s` will be deleted, after migration to `publick8s`

Operations:

- Removed namespace "mirrorbits" to ensure that the PVC `mirrorbits-binary` is removed properly
- Removed the PV `mirrorbits-binary` with `kubectl delete pv mirrorbits-binary` as it's immutable
- Applied manually the Helm chart to re-create the PV, PVC and pod

Checks:
-  Service was back after 2 minutes of outage
- Files are mounted, and mirror redirections are present
- The reported file volume size is now `1000 Gib`

**Comment by dduportal** [comment]
- The performances issues are back when serving the content of `/war` and `/war-stable`
- The access logs shows that we are hammered by GET requests to these to paths

**Comment by dduportal** [comment]
- The `HEADER.html` and `FOOTER.html` files are used by an Apache feature named "FancyIndex": Apache renders the HTML page with these 2 partial templats, and insert in the middle a dynamic directory listing. Ref:
  - https://github.com/jenkinsci/packaging/pull/159
  - https://github.com/jenkinsci/packaging/pull/171

- Checking on the reference VM `pkg.origin.jenkins.io`: 
  - Retrieved locally the 2 files and copying both in the Azure Blob Storage (for the `/war` subdirectory only)
  - Also, the synchronisation log written by the `sync.sh` shows an error: fix published in the commit https://github.com/jenkins-infra/mirror-scripts/commit/bebd7675616a2073173b241ddf5021d098c5f2b4 (fixing up https://github.com/jenkins-infra/mirror-scripts/pull/10 and https://github.com/jenkins-infra/mirror-scripts/pull/11)
  - Triggered the `sync.sh` script as user `mirrorbrain` (as per its crontab)


Let's see the result

**Comment by dduportal** [comment]
- Synchonisation successful
- The Apache errors disappeared with https://github.com/jenkins-infra/kubernetes-management/pull/3863

The performances seems ok for the `/war-stable/` page but not for the `/war`.

**Comment by yashcho** [comment]
@dduportal, Even for me it is the case what do you suggest?

**Comment by dduportal** [comment]
> @dduportal, Even for me it is the case what do you suggest?

I'm not sure to understand, what is the exact problem you are having?

It is true that he page https://get.jenkins.io/war/ is slow, but we do not see HTTP/5XX errors on it.

Do you still see errors are are you blocked by the "slow" response time of this page? If you are block, can you describe what you are trying to achieve (as we could provide alternative and way more efficient solutions).

We are working on this matter but it is not the top priority as nothing seems blocked or broken: happy to reevaluate with more details

**Comment by yashcho** [comment]
> are you blocked by the "slow" response time of this page? 

Yes, basically I am trying to access the URL and fetch the list of versions from table programmatically. So the test cases are failing due to slow response to fetch the table details.

**Comment by dduportal** [comment]
> > are you blocked by the "slow" response time of this page?
> 
> Yes, basically I am trying to access the URL and fetch the list of versions from table programmatically. So the test cases are failing due to slow response to fetch the table details.

In that case, you might be interested by checking the artifactory metadatas. We have an example here, in shell script, which checks for the 2 past LTS releases: https://github.com/jenkinsci/docker/blob/master/.ci/publish.sh#L85-L89 and the 2 past weekly releases: https://github.com/jenkinsci/docker/blob/ca53c743e9e83db0cd235723e6689cd1490e1d13/.ci/publish.sh#L79-L83

**Comment by dduportal** [comment]
> Do you have similar pattern using golang codebase instead of shell script?

No, that way out of our scope. Do not hesitate to propose a contribution if you find or build something.

**Comment by yashcho** [comment]
Sure..

**Comment by dduportal** [comment]
Hello @yashcho , do you still have the problem or can we close this issue?

**Comment by timja** [comment]
This seems far quicker now since https://github.com/jenkins-infra/helpdesk/issues/3917

Plugins list is still slow-ish but does load eventually:
https://get.jenkins.io/plugins/

**Comment by lemeurherve** [comment]
> Plugins list is still slow-ish but does load eventually: https://get.jenkins.io/plugins/

Off-topic & low priority: should an index be generated to avoid Apache having to render this particular folder list like https://serverfault.com/a/843077?
